spec/domain/audits/report_spec.rb

```diff
@@ -0,0 +1,70 @@
+module Audits
+  RSpec.describe Report do
+    before do
+      create(:content_item, title: 'Example')
+    end
+
+    let(:filter) { build(:filter) }
+    let(:url) { 'http://example.com' }
+
+    it 'outputs a header row' do
+      csv = Report.generate(filter, url)
+
+      expect(csv.lines.first).to start_with('Report URL,Report timestamp')
+    end
+
+    it 'outputs whether the item has been audited or not' do
+      csv = Report.generate(filter, url)
+      header = csv.lines.first
+      first_row = csv.lines.third
+
+      expect(header).to match(/Audited/)
+      expect(first_row).to match(/Non audited/)
+    end
+
+    it 'outputs a report metadata row' do
+      Timecop.freeze(2017, 1, 30) do
+        csv = Report.generate(filter, url)
+
+        expect(csv.lines.second).to eq "http://example.com,30/01/17\n"
+      end
+    end
+
+    it 'outputs a row for each content item' do
+      csv = Report.generate(filter, url)
+
+      expect(csv.lines.third).to start_with(',,Example')
+    end
+
+    it "doesn't execute N+1 queries" do
+      ActiveRecord.disable
+
+      expect { Report.generate(filter, url) }.not_to raise_error,
+        'Should not have tried to execute a query after initializing Report'
+    end
+
+    it 'returns all items regardless of the audit status' do
+      create(:passing_audit)
+      create(:failing_audit)
+
+      expect(Report.generate(filter, url).lines.length).to eq(5)
```

Comment:
It's unclear why I'd be expecting 5 rows here. It might be clearer to:

- Contain the item set up in the `before` block in its own separate `describe` block, because it's easy to forget that it exists
- Then write an assertion like `expect(Report.generate(filter, url).lines.length).to eq(%w[metadata_row headers passing_audit failing_audit].count)`